### PR TITLE
chore(Styleguide):  Spring Cleaning

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview

This has been sitting in the backlog a while and since storybook is linking was completely broken I decided to have a hackathon to clean up some of the cruft and knock out the planned design updates.

### Visual Identity => Docs
Brings out Docs Pages into the fold with our visual identity (making it even easier to migrate if we care to, and making our docs a living showcase of gamut).
- Replaces the current DocsWrapper with our own root wrapper.
- Theme overrides and custom mapped components
- Background and ContentContainer standardization.

#### Before
gamut codecademy com_storybook__path=_story_gamut--page
gamut codecademy com_storybook__path=_story_gamut--page (1)

#### After 
localhost_6006__path=_story_gamut--page
localhost_6006__path=_story_gamut--page (1)

### Navigation
Completely refactors the navigation derivation into a single provider and set of methods to look up specific paths along a generated hierarchy with an index / root specification that is configurable at the global level.
- Refactors the base hooks and access pattern for generated links.
- Refactors and uses our own custom `Link` component (LinkTo was consistently broken and unusable).
- Updates components like BreadCrumbs and TableOfContents to uses these new patterns and gamut components exclusively.

### System Prop Addon
A lot of our docgen got completely messed up with Typescript 4.2, since type aliases have an extra layer of caching we cannot get the raw scales from these anymore. 
- Removes a lot of the unnecessary formatting and derivation.
- Refactors to use gamut components.
- Updates all `<Props />` instances to `PropsTable`.

### Other Changes
- Complete reorg and pruning of the `.storybook/components` directory.
- Removes now fully removable stories from the codebase `SplitInterstitial` is dead.

### PR Checklist

- [x] Related to designs:  [Figma](https://www.figma.com/file/TemYhRWOavsERWprDzRJCx/Alerts)
- [x] Related to JIRA ticket: GM-195
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.3935dc.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.a8e845.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.86ac7e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60690715e04a1addc5748155--gamut-preview.netlify.app>https://60690715e04a1addc5748155--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60690715e04a1addc5748155>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.330e71.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606927dde471aa40daeb6a74--gamut-preview.netlify.app>https://606927dde471aa40daeb6a74--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606927dde471aa40daeb6a74>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.871c5c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a5591d0439f60749f0d03--gamut-preview.netlify.app>https://606a5591d0439f60749f0d03--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a5591d0439f60749f0d03>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.df4271.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.0c5952.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a6a4f51f274ac311d01bc--gamut-preview.netlify.app>https://606a6a4f51f274ac311d01bc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a6a4f51f274ac311d01bc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.6b36a6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a6ff4be4e19fd73a6e47b--gamut-preview.netlify.app>https://606a6ff4be4e19fd73a6e47b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a6ff4be4e19fd73a6e47b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.6908e6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a763f51f274ca7a1cff3d--gamut-preview.netlify.app>https://606a763f51f274ca7a1cff3d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a763f51f274ca7a1cff3d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@33.6.1-alpha.959476.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a7ad0ee42b84f05c25d16--gamut-preview.netlify.app>https://606a7ad0ee42b84f05c25d16--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a7ad0ee42b84f05c25d16>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@2.1.1-alpha.eedeaa.0<br/>@codecademy/styleguide@33.6.1-alpha.eedeaa.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a7d4c8b93a444b2662df1--gamut-preview.netlify.app>https://606a7d4c8b93a444b2662df1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a7d4c8b93a444b2662df1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@2.1.1-alpha.fc2f90.0<br/>@codecademy/styleguide@33.6.1-alpha.fc2f90.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606a8421ee42b85e47c25d70--gamut-preview.netlify.app>https://606a8421ee42b85e47c25d70--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606a8421ee42b85e47c25d70>Deploy Logs</a></p>

File: packages/styleguide/stories/Foundations/System/Props.stories.mdx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { formatExternalMarkdown } from '~styleguide/blocks';
     subtitle:
       'All system props accept a syntax that to generate responsive styles on a per prop basis',
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Props.stories.mdx:11]: Just noticed this. While you're here _(and elsewhere)_?

```suggestion
      'All system props accept a syntax to generate responsive styles on a per prop basis',
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@2.1.1-alpha.4a7adb.0<br/>@codecademy/styleguide@33.6.1-alpha.4a7adb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b37df9f5cd52134ffa379--gamut-preview.netlify.app>https://606b37df9f5cd52134ffa379--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b37df9f5cd52134ffa379>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@2.1.1-alpha.928a84.0<br/>@codecademy/styleguide@33.6.1-alpha.928a84.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b41189f76cc37f9c68a65--gamut-preview.netlify.app>https://606b41189f76cc37f9c68a65--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b41189f76cc37f9c68a65>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@2.1.1-alpha.b4e34d.0<br/>@codecademy/styleguide@33.6.1-alpha.b4e34d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b4264263526368d43bd1e--gamut-preview.netlify.app>https://606b4264263526368d43bd1e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b4264263526368d43bd1e>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
Nothing blocking, thanks for taking this on!!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/macros@2.1.1-alpha.ed70c9.0<br/>@codecademy/styleguide@33.6.2-alpha.ed70c9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b4c1fcdafba4175db9b97--gamut-preview.netlify.app>https://606b4c1fcdafba4175db9b97--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b4c1fcdafba4175db9b97>Deploy Logs</a></p>

File: packages/styleguide/.storybook/components/Docs/DocsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  display: grid;
+  background-color: ${themed('colors.white')};
+  padding: ${themed('spacing.16')};
+  grid-column-gap: 2rem;
+  grid-row-gap: 0.5rem;
+`;
+
+export interface GamutParameters extends Parameters {
+  subtitle?: string;
+  status?: 'current' | 'updating' | 'deprecated' | 'unknown' | 'static';
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/styleguide/.storybook/components/Docs/DocsPage.tsx:30]: Love it, was thinking the same thing!

Added the tool tip for this on the pages.

File: packages/styleguide/stories/Foundations/Theme/index.stories.mdx [hunk 3]
```diff
@@ -1,15 +1,15 @@
 import title from '@codecademy/macros/lib/title.macro';
 import { Meta, Story } from '@storybook/addon-docs/blocks';
 
-import { DataTable } from '~styleguide/blocks';
+import { DataTable, Link as LinkTo } from '~styleguide/blocks';
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Theme/index.stories.mdx]: I updated it to LinkTo :) 

Heh yeah it's really annoying.  The iframe communication is the big overhead for clean routing :/

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.7-alpha.4f1d4b.0<br/>@codecademy/gamut-kit@0.2.47-alpha.4f1d4b.0<br/>@codecademy/gamut-labs@13.2.10-alpha.4f1d4b.0<br/>@codecademy/gamut-styles@9.1.4-alpha.4f1d4b.0<br/>@codecademy/gamut-system@0.6.1-alpha.4f1d4b.0<br/>@codecademy/gamut-tests@2.3.4-alpha.4f1d4b.0<br/>@codecademy/gamut@29.7.1-alpha.4f1d4b.0<br/>@codecademy/konamimojisplosion@0.2.4-alpha.4f1d4b.0<br/>@codecademy/macros@2.1.1-alpha.4f1d4b.0<br/>@codecademy/markdown-overrides@0.5.17-alpha.4f1d4b.0<br/>@codecademy/styleguide@33.6.2-alpha.4f1d4b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606b5692d3ef3449f085fe16--gamut-preview.netlify.app>https://606b5692d3ef3449f085fe16--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606b5692d3ef3449f085fe16>Deploy Logs</a></p>

**Pull request merged**